Gems/Multiplayer/Code/Include/Multiplayer/Components/SimplePlayerSpawnerComponentBus.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/EBus/EBus.h>
+
+
+namespace Multiplayer
+{
+    //! @class SimplePlayerSpawnerRequests
+    //! @brief The SimplePlayerSpawnerRequest event-bus exposes helper methods regarding network player spawners.
+    //! For example, it's common for game servers want information regarding valid spawn locations in order to reset a player's position after the player dies or the round restarts (respawning).
+    class SimplePlayerSpawnerRequests : public AZ::EBusTraits
+    {
+    public:
+        static const AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Single;
+        static const AZ::EBusAddressPolicy AddressPolicy = AZ::EBusAddressPolicy::Single;
+        
+        //! Visits and returns the next player spawn point, causing the current spawn point index to increment.
+        //! Method is only valid if called from the multiplayer host/authority; clients are not given information regarding spawn points.
+        //! @return AZ::Transform The location of the next spawn point
+        virtual AZ::Transform VisitNextSpawnPoint() = 0;
+
+        //! Returns the next player spawn point. Unlike VisitNextSpawnPoint, this will not cause the current spawn point index to increment.
+        //! Method is only valid if called from the multiplayer host/authority; clients are not given information regarding spawn points.
+        //! @return AZ::Transform The location of the next spawn point
+        virtual AZ::Transform GetNextSpawnPoint() const = 0;
```

Comment:
Just curious about the API here. Is there any particular reason for exposing it this way, vs something like:

int GetNumSpawnPoints()
Transform GetSpawnPoint(int index)
int GetCurrentSpawnPointIndex();
void SetCurrentSpawnPointIndex(int index);
// And maybe some APIs for adding / removing / changing spawn point locations too?

The API just surprised me on first read, since there's nothing that exposes the set of spawn points as a whole, and the naming difference between "VisitNextSpawnPoint" vs "GetNextSpawnPoint" is a little bit subtle.

I'm not entirely sure what the intended uses are for the APIs as written, it seems like they're intended to maybe enable spawning something other than players? If so, maybe the API should be SpawnAsset(AssetId) for spawning something other than players, or SpawnPlayer(xxx), or something?

Reply:
The API and component is inspired by the Multiplayer Sample game for respawning. The MPS custom spawner calls a "VisitNextSpawnPoint" method (actually just confusingly called GetNextSpawnPoint but it updates the spawn index internally).

Reply:
Visit* is a bit of a curious name. It doesn't make it clear enough that something is modified behind the scenes. Maybe something along the lines of Get and Increment? Or something. Hmm.

Reply:
Hm, what about:
```
AZ::Transform AdvanceSpawnPoint(); // advances the spawn point index then returns the new spawn point
AZ::Transform GetCurrentSpawnPoint() const; // gets the current spawn point that pointed at
```

Reply:
Oops, just saw your naming suggestions. I changed the name to GetNextSpawnPoint vs RoundRobinNextSpawnPoint. Does that make it more distinguished? 

I like using "next" vs "current" since spawning refers to a future action, as opposed to an ongoing action.

Reply:
Changed to "Advance" thanks!